File: C#/Transformer/CharTransformer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using XmlBinder.Exception;

namespace XmlBinder.Transformer
{
	public class CharTransformer : XmlBinder.Interface.TransformerInterface
	{
		public static CharTransformer Instance = new CharTransformer();

		public object read(string _value)
		{
			if (_value.Length != 1)
			{
				throw new TransformerException("CharTransformer", "read", "too many character in string to convert in char");
			}
			return _value[0];
		}

		public string write(object _value)
		{
			if (_value is char)
			{
				return "" + (char)_value;
			}
			else
			{
				throw new TransformerException("CharTransformer", "write", "the object of type " + _value.GetType().ToString() + " is not a char");
			}
		}
	}
}

File: C++/XmlBinder/Transformer/chartransformer.cpp
#include "chartransformer.hpp"

CharTransformer * CharTransformer::instance = new CharTransformer();

CharTransformer::CharTransformer()
{
}


QVariant CharTransformer::read(QString _value)
{
	if(_value.size() <= 0)
	{
		throw QString("Can not convert value %1 to char").arg(_value);
	}
	return _value.toStdString().at(0);
}

QString CharTransformer::write(const QVariant & _value)
{
	if(_value.canConvert<char>())
	{
		return QString("%1").arg(_value.value<char>());
	}
	throw QString("Can not convert value to char");
}
